File: 0525-contiguous-array/0525-contiguous-array.java
class Solution {
    public int findMaxLength(int[] nums) {
        int maxlen =0; 
        int sum =0; 
        Map<Integer,Integer>mp = new HashMap<>(); 
        mp.put(0,-1);

        for( int i =0 ; i< nums.length; i++){
            sum+=(nums[i] == 0)? -1 : 1 ; 

            if(mp.containsKey(sum)) maxlen = Math.max(maxlen,i - mp.get(sum));
            else mp.put(sum,i);
        }
        return maxlen;
    }
}

File: 0525-contiguous-array/0525-contiguous-array.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        unordered_map<int,int>mp; 
        mp[0] = -1; 
        int maxlen = 0; 
        int sum = 0; 

        for( int i =0 ; i<nums.size() ; i++){
            sum+=(nums[i] == 0 ) ? -1 : 1;

            if(mp.find(sum) != mp.end()) maxlen = max(maxlen,i-mp[sum]); 
            else  mp[sum] = i;
        }
        return maxlen ;
    }
};